pymc3/step_methods/smc.py

```diff
@@ -177,16 +188,18 @@ def _initial_population(draws, model, variables):
     """
     Create an initial population from the prior
     """
+
     population = []
     var_info = {}
     start = model.test_point
+    dict_to_array = DictToArrayBijection(ArrayOrdering(variables), start).map
     init_rnd = pm.sample_prior_predictive(draws, model=model)
     for v in variables:
         var_info[v.name] = (start[v.name].shape, start[v.name].size)
 
     for i in range(draws):
         point = pm.Point({v.name: init_rnd[v.name][i] for v in variables}, model=model)
-        population.append(model.dict_to_array(point))
+        population.append(dict_to_array(point))
```

Comment:
Where is the actual fix here?

Reply:
`model.dict_to_array` returns an empty array for discrete variables (bijection method is reading only continuous variables). Thus I basically create a version that reads all variables.

Reply:
Isn't that bug then upstream of this? Why should `model.dict_to_array` only return continuous variables in the first place?

Reply:
not sure, I ask that in slack.

Reply:
@twiecki what do you think we should do?